pkg/netdevice/netResourcePool_test.go

```diff
@@ -237,4 +237,107 @@ var _ = Describe("NetResourcePool", func() {
 			})
 		})
 	})
+	Describe("Health Checking on Net Devices", func() {
+		Context("PF link status and device existance for VFs with Same PF", func() {
+			rf := factory.NewResourceFactory("fake", "fake", true, false)
+			nadutils := rf.GetNadUtils()
+			rc := &types.ResourceConfig{
+				ResourceName:             "fake",
+				ResourcePrefix:           "fake",
+				CheckHealthOnPf:          true,
+				CheckHealthOnDeviceExist: true,
+				SelectorObjs:             []interface{}{&types.NetDeviceSelectors{}},
+			}
+
+			var fake1health bool = true
+			var fake2health bool = true
+
+			fake1 := &mocks.PciNetDevice{}
+			fake1.On("GetPfNetName").Return("fakepf1")
+			fake1.On("GetPciAddr").Return("0000:01:00.1")
+			fake1.On("SetHealth", Anything).Run(func(args Arguments) {
+				fake1health = args.Bool(0)
+			}).Return()
+
+			fake2 := &mocks.PciNetDevice{}
+			fake2.On("GetPfNetName").Return("fakepf1")
+			fake2.On("GetPciAddr").Return("0000:01:00.2")
+			fake2.On("SetHealth", Anything).Run(func(args Arguments) {
+				fake2health = args.Bool(0)
+			}).Return()
+
+			pcis := map[string]types.HostDevice{"fake1": fake1, "fake2": fake2}
+
+			rp := netdevice.NewNetResourcePool(nadutils, rc, pcis)
+
+			SetCurrentHealth := func(health bool) {
+				fake1health = health
+				fake2health = health
+				fake1.On("GetHealth").Unset()
+				fake2.On("GetHealth").Unset()
+				fake1.On("GetHealth").Return(health).Once()
+				fake2.On("GetHealth").Return(health).Once()
+			}
+
+			SetCurrentLinkState := func(up bool) {
+				fake1.On("IsPfLinkUp").Unset()
+				fake2.On("IsPfLinkUp").Unset()
+				fake1.On("IsPfLinkUp").Return(up, nil).Once()
+				fake2.On("IsPfLinkUp").Return(up, nil).Once()
+			}
+
+			SetCurrentDeviceExistance := func(exist bool) {
+				fake1.On("DeviceExists").Unset()
+				fake2.On("DeviceExists").Unset()
+				fake1.On("DeviceExists").Return(exist, nil).Once()
+				fake2.On("DeviceExists").Return(exist, nil).Once()
+			}
+
+			RunTest := func(health, link, exist bool) {
```

Comment:
we can leverage the ginkgo DescribeTable feature 
https://onsi.github.io/ginkgo/#table-specs

or we can repeat the logic a little in each of the `It()` test case, removing the `if/else` statement from the assertion. Though this involves repeating the code, it makes the test cases linear and readable. 

Up to you which solution to adopt

Reply:
I think it makes sense to repeat the logic a little in each It() case. This was what I had it before. I cleaned it up now.